// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTRPCEnums";
option optimize_for = LITE_RUNTIME;

enum RPCCommandType {
    RPCCOMMAND_ERROR = 0;
    RPCCOMMAND_ADDCLIENTSESSION = 1;
    RPCCOMMAND_ADDSERVERSESSION = 2;
    RPCCOMMAND_LISTCLIENTSESSIONS = 3;
    RPCCOMMAND_LISTSERVERSESSIONS = 4;
    RPCCOMMAND_IMPORTHDSEED = 5;
    RPCCOMMAND_LISTHDSEEDS = 6;
    RPCCOMMAND_GETHDSEED = 7;
    RPCCOMMAND_CREATENYM = 8;
    RPCCOMMAND_LISTNYMS = 9;
    RPCCOMMAND_GETNYM = 10;
    RPCCOMMAND_ADDCLAIM = 11;
    RPCCOMMAND_DELETECLAIM = 12;
    RPCCOMMAND_IMPORTSERVERCONTRACT = 13;
    RPCCOMMAND_LISTSERVERCONTRACTS = 14;
    RPCCOMMAND_REGISTERNYM = 15;
    RPCCOMMAND_CREATEUNITDEFINITION = 16;
    RPCCOMMAND_LISTUNITDEFINITIONS = 17;
    RPCCOMMAND_ISSUEUNITDEFINITION = 18;
    RPCCOMMAND_CREATEACCOUNT = 19;
    RPCCOMMAND_LISTACCOUNTS = 20;
    RPCCOMMAND_GETACCOUNTBALANCE = 21;
    RPCCOMMAND_GETACCOUNTACTIVITY = 22;
    RPCCOMMAND_SENDPAYMENT = 23;
    RPCCOMMAND_MOVEFUNDS = 24;
    RPCCOMMAND_ADDCONTACT = 25;
    RPCCOMMAND_LISTCONTACTS = 26;
    RPCCOMMAND_GETCONTACT = 27;
    RPCCOMMAND_ADDCONTACTCLAIM = 28;
    RPCCOMMAND_DELETECONTACTCLAIM = 29;
    RPCCOMMAND_VERIFYCLAIM = 30;
    RPCCOMMAND_ACCEPTVERIFICATION = 31;
    RPCCOMMAND_SENDCONTACTMESSAGE = 32;
    RPCCOMMAND_GETCONTACTACTIVITY = 33;
    RPCCOMMAND_GETSERVERCONTRACT = 34;
    RPCCOMMAND_GETPENDINGPAYMENTS = 35;
    RPCCOMMAND_ACCEPTPENDINGPAYMENTS = 36;
    RPCCOMMAND_GETCOMPATIBLEACCOUNTS = 37;
    RPCCOMMAND_CREATECOMPATIBLEACCOUNT = 38;
    RPCCOMMAND_GETWORKFLOW = 39;
    RPCCOMMAND_GETSERVERPASSWORD = 40;
    RPCCOMMAND_GETADMINNYM = 41;
    RPCCOMMAND_GETUNITDEFINITION = 42;
    RPCCOMMAND_GETTRANSACTIONDATA = 43;
    RPCCOMMAND_LOOKUPACCOUNTID = 44;
    RPCCOMMAND_RENAMEACCOUNT = 45;
}

enum RPCResponseCode {
    RPCRESPONSE_INVALID = 0;
    RPCRESPONSE_SUCCESS = 1;
    RPCRESPONSE_BAD_SESSION = 2;
    RPCRESPONSE_NONE = 3;
    RPCRESPONSE_QUEUED = 4;
    RPCRESPONSE_UNNECESSARY = 5;
    RPCRESPONSE_RETRY = 6;
    RPCRESPONSE_NO_PATH_TO_RECIPIENT = 7;
    RPCRESPONSE_BAD_SERVER_ARGUMENT = 8;
    RPCRESPONSE_CHEQUE_NOT_FOUND = 9;
    RPCRESPONSE_PAYMENT_NOT_FOUND = 10;
    RPCRESPONSE_START_TASK_FAILED = 11;
    RPCRESPONSE_NYM_NOT_FOUND = 12;
    RPCRESPONSE_ADD_CLAIM_FAILED = 13;
    RPCRESPONSE_ADD_CONTACT_FAILED = 14;
    RPCRESPONSE_REGISTER_ACCOUNT_FAILED = 15;
    RPCRESPONSE_BAD_SERVER_RESPONSE = 16;
    RPCRESPONSE_WORKFLOW_NOT_FOUND = 17;
    RPCRESPONSE_UNITDEFINITION_NOT_FOUND = 18;
    RPCRESPONSE_SESSION_NOT_FOUND = 19;
    RPCRESPONSE_CREATE_NYM_FAILED = 20;
    RPCRESPONSE_CREATE_UNITDEFINITION_FAILED = 21;
    RPCRESPONSE_DELETE_CLAIM_FAILED = 22;
    RPCRESPONSE_ACCOUNT_NOT_FOUND = 23;
    RPCRESPONSE_MOVE_FUNDS_FAILED = 24;
    RPCRESPONSE_REGISTER_NYM_FAILED = 25;
    RPCRESPONSE_CONTACT_NOT_FOUND = 26;
    RPCRESPONSE_ACCOUNT_OWNER_NOT_FOUND = 27;
    RPCRESPONSE_SEND_PAYMENT_FAILED = 28;
    RPCRESPONSE_TRANSACTION_FAILED = 29;
    RPCRESPONSE_TXID = 30;
    RPCRESPONSE_UNIMPLEMENTED = 254;
    RPCRESPONSE_ERROR = 255;
}

enum RPCPushType {
    RPCPUSH_ERROR = 0;
    RPCPUSH_ACCOUNT = 1;
    RPCPUSH_CONTACT = 2;
    RPCPUSH_TASK = 3;
}

enum RPCPaymentType {
    RPCPAYMENTTYPE_ERROR = 0;
    RPCPAYMENTTYPE_CHEQUE = 1;
    RPCPAYMENTTYPE_TRANSFER = 2;
    RPCPAYMENTTYPE_VOUCHER = 3;
    RPCPAYMENTTYPE_INVOICE = 4;
    RPCPAYMENTTYPE_BLINDED = 5;
    RPCPAYMENTTYPE_BLOCKCHAIN = 6;
}

enum AccountEventType {
    ACCOUNTEVENT_ERROR = 0;
    ACCOUNTEVENT_INCOMINGCHEQUE = 1;
    ACCOUNTEVENT_OUTGOINGCHEQUE = 2;
    ACCOUNTEVENT_INCOMINGTRANSFER = 3;
    ACCOUNTEVENT_OUTGOINGTRANSFER = 4;
    ACCOUNTEVENT_INCOMINGINVOICE = 5;
    ACCOUNTEVENT_OUTGOINGINVOICE = 6;
    ACCOUNTEVENT_INCOMINGVOUCHER = 7;
    ACCOUNTEVENT_OUTGOINGVOUCHER = 8;
    ACCOUNTEVENT_INCOMINGBLOCKCHAIN = 9;
    ACCOUNTEVENT_OUTGOINGBLOCKCHAIN = 10;
}

enum ContactEventType {
    CONTACTEVENT_ERROR = 0;
    CONTACTEVENT_INCOMINGMESSAGE = 1;
    CONTACTEVENT_OUTGOINGMESSAGE = 2;
    CONTACTEVENT_INCOMONGPAYMENT = 3;
    CONTACTEVENT_OUTGOINGPAYMENT = 4;
}

enum AccountType {
    ACCOUNTTYPE_ERROR = 0;
    ACCOUNTTYPE_NORMAL = 1;
    ACCOUNTTYPE_ISSUER = 2;
    ACCOUNTTYPE_BLOCKCHAIN = 3;
}
